bin/porta

```diff
@@ -626,10 +688,26 @@ module Porta
   end
 
   class ResetCommand < CommandRunner
+    include Apisonator
+
     def run
+      reset_redis
+      reset_system
+      reset_zync
+    end
+
+    def reset_redis
       system('redis-cli flushall') || raise("redis-cli execution failed")
-      in_porta(:system, porta_additional_envs) { "bundle exec rails db:reset" } || raise("db:reset in Porta failed")
-      in_zync {"bundle exec rails db:reset" } || raise("db:reset in Zync failed")
+    end
+
+    def reset_system
+      run_apisonator_listener && in_porta(:system, porta_additional_envs) { "bundle exec rails db:reset" } || raise("db:reset in Porta failed")
+    ensure
+      stop_apisonator_listener
```

Comment:
what if it was already working before? this will kill it a force the user to manually launch it after. Or am I wrong?

Reply:
I don't understand why `porta reset` needs apisonator to be up. If it can be replaced by the fake server, that's better because it'll remove a dependency.

On the other hand, it usually happens to me that I run `porta reset` and the `db:seed` step fails because some of the deps are up or down when they shouldn't. For instance, I usually see an error complaining about zync already running. It's not clear for me what should and shouldn't be running when in launch `porta reset`. I think the dependencies for this command should be either enforced in the code or documented in the its help dialog.

Reply:
Steps are documented in the porta-dev-tools readme at the bottom. Enforcing order of commands is outside the scope of this PR.

When you configure fake server, then the changes are not reflected in the real apisonator so API calls wont work. That's why changed porta config to use real apisonator that porta-dev-tools starts. But issue arises on reset because porta tries to contact apisonator during db seed. But apisonator should be down for the reset or redis. This is about making things work simple out of the box.

if apisonator is running during db:seed, then it will pick-up the changes and `resync` will not be needed later. So in general doing what this PR does is more natural than using fake server. But it is a little complicated so I'm considering to always use fake server enforced in porta code, during `db:seed` and then user will need to `resync` backend.

More clear?

Reply:
yes, more clear. Why not then calling also `stop_apisonator_listener` from `reset_redis` that way apisonator would be down for redis and up for system.

Reply:
This fixes the issue on porta side: 3scale/porta#3283

Also will not hurt any use case we want to implement here.

Reply:
Not required, only "if we want". For example to avoid the `resync` need.

Reply:
Then the `README` should be updated too